src/Tools/babylon.assetsManager.ts

```diff
@@ -669,6 +669,118 @@ module BABYLON {
+        private _framesData: any = {};
+        /**
+         * Gets data loaded
+         */
+        public data: any = {};
+        /**
+         * Callback called when the task is successful
+         */
+        public onSuccess: (task: ImageAssetTask) => void;
+        /**
+         * Callback called when the task is successful
+         */
+        public onError: (task: ImageAssetTask, message?: string, exception?: any) => void;
+
+        /**
+         * Creates a new AtlasTextureAssetTask
+         * @param name defines the name of the task
+         * @param url defines the location of the atlas texture to load
+         */
+        constructor(public name: string, public url: string) {
+            super(name);
+        }
+
+        /**
+         * Execute the current task
+         * @param scene defines the scene where you want your assets to be loaded
+         * @param onSuccess is a callback called when the task is successfully executed
+         * @param onError is a callback called if an error occurs
+         */
+        public runTask(scene: Scene, onSuccess: () => void, onError: (message?: string, exception?: any) => void) {
+            let loadJsonFile = () => {
+                scene._loadFile(this.url, (data) => {
+                    let jsonData = JSON.parse(data as string);
+                    if (jsonData.meta) {
+                        this._framesData = jsonData.frames;
+                        let imageName = jsonData.meta.image;
+                        let parentPath = Tools.GetFolderPath(this.url);
+                        loadImageFile(parentPath + imageName);
+                    } else
+                        Tools.Error('Not found meta tag in JSON file: ' + this.url);
+                }, undefined, false, false, (request, exception) => {
+                    if (request) {
+                        onError(request.status + " " + request.statusText, exception);
+                    }
+                });
+            };
+
+            let loadImageFile = (url: string) => {
+                Tools.LoadImage(url, (img) => {
+                    this._parseData(img);
+                    onSuccess();
+                }, function (message, error) {
+                    onError(message, error);
+                }, null);
+            };
+
+            loadJsonFile();
+        }
+
+        /**
+         * Data parsed when load success JSON and Image file
+         * @param {HTMLImageElement} img
+         * @private
+         */
+        private _parseData(img: HTMLImageElement) {
+            Object.keys(this._framesData).forEach((key) => {
+                let name = key.substring(0, key.lastIndexOf('.'));
+                this.data[name] = new AtlasTexture(name, img, this._framesData[key]);
+            });
+        }
+    }
+
+    /**
+     * Defines the interface used storage information atlas texture
+     */
+    export interface IAtlasTexture {

```

Comment:
As this is specific to GUI it should be in GUI

Reply:
OK!